src/systems/weapons/player_weapons/player_blaster/player_blaster_collision.rs

```diff
@@ -1,22 +1,35 @@
-use crate::queries::{
-    player_blaster_queries::{MutablePlayerBlasterEntityTransformQuery, PlayerBlasterFilter},
-    starship_queries::{MutableStarshipEntityTransformQuery, StarshipFilter},
-};
-use bevy::{
-    asset::AssetServer,
-    audio::{AudioBundle, PlaybackMode, PlaybackSettings, Volume},
-    ecs::system::Res,
-    prelude::{Commands, Query},
-    utils::tracing,
+use crate::{
+    events::{
+        collision_events::PlayerBlasterCollisionEvent, despawn_sprite_event::DespawnSpriteEvent,
+    },
+    queries::{
+        player_blaster_queries::{MutablePlayerBlasterEntityTransformQuery, PlayerBlasterFilter},
+        starship_queries::{MutableStarshipTransformQuery, StarshipFilter},
+    },
 };
+use bevy::{ecs::event::EventWriter, prelude::Query};
 
 // TODO multi-thread
 pub fn player_blaster_collision_with_starship(
-    mut commands: Commands,
-    asset_server: Res<AssetServer>,
     mut player_blasters: Query<MutablePlayerBlasterEntityTransformQuery, PlayerBlasterFilter>,
-    mut starships: Query<MutableStarshipEntityTransformQuery, StarshipFilter>,
+    mut starships: Query<MutableStarshipTransformQuery, StarshipFilter>,
+    mut player_blaster_collision_event: EventWriter<PlayerBlasterCollisionEvent>,
+    mut despawn_sprite_event: EventWriter<DespawnSpriteEvent>,
 ) {
+    // TODO Multi-thread this way requires clone
```

Comment:
This would be memory intensive as each "starship" would need a memory copy

Reply:
Maybe this is telling... and it should just be left as single threaded

Reply:
I doubt we're going to have that many collisions happening in one frame so single threaded is probably fine for now